src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexPrefixAnalyzer.cs

```diff
@@ -865,6 +896,19 @@ private static RegexNodeKind FindLeadingOrTrailingAnchor(RegexNode node, bool le
             }
         }
 
+        private static float SumFrequencies(ReadOnlySpan<char> chars)
+        {
+            float sum = 0;
+            foreach (char c in chars)
+            {
+                // Lookup each character in the table.  For values > 255, this will end up truncating
+                // and thus we'll get skew in the data.  It's already a gross approximation, though,
+                // and it is primarily meant for disambiguation of ASCII letters.
+                sum += s_frequency[(byte)c];
```

Comment:
The last 128 floats in the table are all zero -- would it be worth an "if ((byte)c <= 0x7f) " here so that could be cut off and save 512 bytes?

Reply:
or even `&& (byte)c > 0x1f` as well for another 192 bytes.

Reply:
I'd rather augment the data sources with non-ascii.

Reply:
final thought, you could halve again by storing shorts instead of floats. with all this, you drop from 1024 bytes to 192 bytes.